plugins/catalog-backend-module-ldap/src/ldap/client.ts

```diff
@@ -44,9 +46,28 @@ export class LdapClient {
     bind?: BindConfig,
     tls?: TLSConfig,
   ): Promise<LdapClient> {
+    const readTLSOptionFile = (file?: string) =>
+      file !== undefined ? fs.readFileSync(file).toString() : undefined;
+
+    const getTlsOptions = () => {
+      const certs = readTLSOptionFile(tls?.certs);
+      const keys = readTLSOptionFile(tls?.keys);
+
+      if (certs !== undefined || keys !== undefined) {
+        return {
+          secureContext: tlsLib.createSecureContext({
+            cert: certs,
+            key: keys,
+          }),
+          rejectUnauthorized: tls?.rejectUnauthorized,
+        };
+      }
+      return tls;
+    };
```

Comment:
This can be simplified :)

Reply:
I've tried to simplify it in next commit. Hopefully that improves it

Reply:
You could remove some `?` & `undefined` checks by just wrapping the relevant code in a `tls && tls.certs && tls.key` check. If `tls.certs` & `tls.keys` are undefined, `readTLSOptionFile` wouldn't run & therefor wouldn't need to be defined.

Reply:
Thanks. Sorry about the verbosity - I work with JS/TS only for hobby projects so not really adept at writing good quality TS code. But I hope the code is in better shape now